src/frontend/js/hooks/useEnroll/index.spec.tsx

```diff
@@ -39,17 +39,24 @@ describe('useEnroll ( edge case )', () => {
     );
   };
 
+  beforeEach(() => {
+    // SessionProvider inital requests
+    fetchMock.get('https://joanie.endpoint/api/v1.0/orders/', []);
+    fetchMock.get('https://joanie.endpoint/api/v1.0/addresses/', []);
+    fetchMock.get('https://joanie.endpoint/api/v1.0/credit-cards/', []);
+  });
+
   afterEach(() => {
     jest.clearAllMocks();
     fetchMock.restore();
   });
 
   it('should return early when a course run is not enrollable', async () => {
+    fetchMock.get('https://joanie.endpoint/api/v1.0/enrollments/', []);
     fetchMock.post('https://joanie.endpoint/api/v1.0/enrollments/', {});
     const { result } = renderHook(() => useEnroll([], undefined), { wrapper });
-    await waitFor(() => {
-      expect(result.current?.enroll).toBeDefined();
-    });
+    await waitFor(() => expect(result.current.isLoading).toBeFalsy());
```

Comment:
Stricter is better. Prefer `toBe` to prevent false positive tests.
```suggestion
    await waitFor(() => expect(result.current.isLoading).toBe(false));
```